generator/src/test/java/com/scottlogic/deg/generator/inputs/JsonProfileReaderTests.java

```diff
@@ -179,6 +179,46 @@ public void shouldReadNameOfNamedRules() throws IOException, InvalidProfileExcep
                 ruleWithDescription("Too rule for school"));
     }
 
+    @Test
+    public void shouldNotThrowIsNullWithValueNull() {
+        givenJson(
+            "{" +
+                "    \"schemaVersion\": \"v3\"," +
+                "    \"fields\": [ { \"name\": \"foo\" } ]," +
+                "    \"rules\": [" +
+                "        {" +
+                "           \"rule\": \"Too rule for school\"," +
+                "           \"constraints\": [" +
+                "               { \"field\": \"foo\", \"is\": \"null\", \"value\": null }" +
+                "           ]" +
+                "        }" +
+                "    ]" +
+                "}");
+
+        Assertions.assertDoesNotThrow(
+            () -> new InvalidProfileException("Couldn't recognise 'value' property, it must be set to a value"));
+    }
+
+    @Test
+    public void shouldNotThrowIsNullWithValuesNull() {
+        givenJson(
+            "{" +
+                "    \"schemaVersion\": \"v3\"," +
+                "    \"fields\": [ { \"name\": \"foo\" } ]," +
+                "    \"rules\": [" +
+                "        {" +
+                "           \"rule\": \"Too rule for school\"," +
+                "           \"constraints\": [" +
+                "               { \"field\": \"foo\", \"is\": \"null\", \"values\": null }" +
+                "           ]" +
+                "        }" +
+                "    ]" +
+                "}");
+
+        Assertions.assertDoesNotThrow(
```

Comment:
Currently this proves that the call to the constructor of the exception doesn't throw anything. We need to assert that the profile can be read without exception, therefore the `read` method on `ProfileReader` needs to be exercised.

Is suspect you need something like

```
Assert.assertDoesNotThrow(() -> getResultingProfile());
```